dotnet/src/SemanticKernel.Core/Events/SKEventArgsExtensions.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+#pragma warning disable IDE0130
+// ReSharper disable once CheckNamespace - Using the main namespace
+namespace Microsoft.SemanticKernel.Events;
+
+/// <summary>
+/// SKEventArgs extensions
+/// </summary>
+public static class SKEventArgsExtensions
+{
+    /// <summary>
+    /// Only tries to get the rendered prompt from the event args metadata if it exists.
+    /// </summary>
+    /// <param name="eventArgs">Target event args to extend</param>
+    /// <param name="renderedPrompt">Outputs the renderedPrompt</param>
+    /// <returns>True if the prompt was present</returns>
+    public static bool TryGetRenderedPrompt(this SKEventArgs eventArgs, out string? renderedPrompt)
+    {
+        var found = eventArgs.Metadata.TryGetValue(SemanticFunction.RenderedPromptMetadataKey, out var renderedPromptObject);
```

Comment:
Is `SemanticFunction` the right place to store the key name? That class is internal so I want to write my own function to pull values out of the Metadata I don't have an easy way to discover the keys.

Reply:
I see, one way would be to add that constant to the SKEventArgs class itself, as this applies to the Metadata dictionary of it. What you think?

Reply:
Agreed in moving this to SkEventArgs class.